packages/wallets/src/components/AccountsList/__tests__/AccountsList.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ComponentProps, PropsWithChildren } from 'react';
+import { APIProvider } from '@deriv/api';
+import { render, screen } from '@testing-library/react';
+import useDevice from '../../../hooks/useDevice';
+import { ModalProvider } from '../../ModalProvider';
+import WalletMobileTourGuide from '../../WalletTourGuide/WalletMobileTourGuide';
+import AccountsList from '../AccountsList';
+
+jest.mock('../../../hooks/useDevice');
+const mockUseDevice = useDevice as jest.MockedFunction<typeof useDevice>;
+
+const mockWalletMobileTourGuide = jest.fn();
+jest.mock(
+    '../../WalletTourGuide/WalletMobileTourGuide',
+    // eslint-disable-next-line react/display-name
+    () => (props: ComponentProps<typeof WalletMobileTourGuide>) => {
+        mockWalletMobileTourGuide(props);
+        return <p>mock wallet tour guide</p>;
```

Comment:
Instead of spying, its typically easier to display specific props. Displaying them will also catch an intermittend scenarios when component is called with correct data and then again with incorrect (and we do have lots of race hazards like that)

e.g:

`jest.mock(
    '../../WalletTourGuide/WalletMobileTourGuide',
    () => (props: ComponentProps<typeof WalletMobileTourGuide>) => {
        return (
        <p>
          <span>mock wallet tour guide</span>
          <span>{isMT5PlatformListLoaded ? 'mt5 loaded prop' : 'mt5 not loaded prop'}</span>
          /* other props displayed one way or the other */
          
        
        </p>
        );
        }